### DISCUSSION TITLE
TypeError: The 'compilation' argument must be an instance of Compilation

### DISCUSSION THREAD

**Discussion by popmanhe** [vercel/next.js#22407] [Help]
**What example does this report relate to?**

with-ant-design-less

**What version of Next.js are you using?**

10.0.7

**What version of Node.js are you using?**

14

**What browser are you using?**

chrome

**What operating system are you using?**

windows 10

**How are you deploying your application?**

local

**Describe the Bug**

If upgrade webpack to v5, I got the error: TypeError: The 'compilation' argument must be an instance of Compilation.

It is working for webpack v4.44.2

**Expected Behavior**

Works with webpack v5

**To Reproduce**

1. Install "webpack": "5.23.0" in package.json
 2. Run "npm run dev"

**Comment by nazywamsiepawel** [comment]
+1, same issue here. Webpack 5.23.0 and next 10.0.3

**Comment by viniciusclerod** [comment]
+1, same issue here. Webpack 5.23.0 and next 10.0.7

**Comment by umarmw** [comment]
+1, same issue here. Webpack 5.24.0 and next 10.0.7

**Comment by hridoy43** [comment]
facing the same issue with next.js 10.0.7 and webpack 5.24.1

**Comment by thekcsam** [comment]
facing the same issue with next.js 10.0.7 and webpack 5.24.2

Might be related to [this](https://github.com/jantimon/html-webpack-plugin/issues/1451#issuecomment-717058113)

**Comment by watch-janick** [comment]
Anyone has any idea how to resolve this error?

**Comment by vincegio** [comment]
I find it weird that it has been bounced over to "discussions", it feels like this is a (rather big) issue.
Haven't found a solution yet...

**Comment by pwalczyszyn** [comment]
I got it working by installing webpack@4

**Reply by vincegio** [reply]
Appears to work, surprisingly. next-less might be outdated then? Hasn't really been touched in a while.

**Comment by hridoy43** [comment]
Guys, I have just made [this boilerplate](https://github.com/hridoy43/next-antd-boilerplate) of **nextjs and ant design (with less support)**.  
**Webpack v5** works well with this boilerplate locally but can't deploy it on vercel for "Critters" related issues (webpack error). 
You can fork and test this.

(Also, if you are facing any issue with less support in nextjs then you can use this boilerplate. it's working out of the box with zero configuration)

**Comment by vincegio** [comment]
Following this reply above, I checked why it worked and https://github.com/SolidZORO/next-plugin-antd-less might actually be the answer. It has "Zero Dependency on other Next-Plugins..".

Set it up, uninstalled other packages that was used in my old config but had to explicitly install less-loader@7.3.0 (same version as the plugin uses) for it to work. Can also confirm webpack 5 works.

**Reply by yaolei** [reply]
could you share your package.json file ?

**Comment by umarmw** [comment]
I found out the root cause for mine. There were two different versions of webpack installed; webpack 4 by next-offline and webpack 5 added in the package.json file. 

To know which modules are using webpack and/or its version, run the following command: 
`npm ls webpack`

The next issue what that I was using zeit/next-css and zeit/next-sass which is now deprecated as NextJS support them natively.

There were also some CSS files referenced in some reusable components. They have been moved to the pages/_app.tsx file.

This is how I've fixed mine. 
Hope some steps might be useful to you guys.

**Comment by yaolei** [comment]
Is anyone have resolve this issue or is there any solution for it ?
